lcmark.lua

```diff
@@ -354,21 +369,23 @@ function lcmark.convert(inp, to, options)
   if not writer then
     return nil, nil, ("Unknown output format " .. tostring(to))
   end
-  local opts, columns, filters, yaml_metadata
+  local opts, columns, filters, yaml_metadata, yaml_parser
   if options then
      opts = toOptions(options)
      columns = options.columns or 0
      filters = options.filters or {}
      yaml_metadata = options.yaml_metadata
+     yaml_parser = options.yaml_parser or default_yaml_parser
   else
      opts = cmark.OPT_DEFAULT
      columns = 0
      filters = {}
      yaml_metadata = false
+     yaml_parser = default_yaml_parser
   end
   local doc, meta
   if yaml_metadata then
-    doc, meta = parse_document_with_metadata(inp, opts)
+    doc, meta = parse_document_with_metadata(inp, yaml_parser, opts)
```

Comment:
It would be more compatible with the earlier API if we made `yaml_parser` the third parameter instead of the second.  That way, existing uses would simply assign it nil and would continue to work.

Another option could be to put the `yaml_parser` into `opts`?

Reply:
> Another option could be to put the `yaml_parser` into `opts`?

I'd love to, but `opts` are the cmark options, packed into an integer.

> It would be more compatible with the earlier API if we made `yaml_parser` the third parameter instead of the second. That way, existing uses would simply assign it nil and would continue to work.

Compatibility isn't an issue here, as `parse_document_with_metadata` is a local function and thus can't be called by the user. I put the `yaml_parser` parameter before `opts` simply because it felt more "significant" (if that makes any sense) but at the end of the day, it doesn't really matter.

Reply:
ah, okay, thanks for clarifying.